solc/CommandLineInterface.cpp

```diff
@@ -1097,6 +1097,21 @@ void CommandLineInterface::outputCompilationResults()
 {
 	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
 
+	std::set<std::string> uncompiledContracts;
+	for (auto const& contract: m_compiler->contractNames())
+		if (objectWithLinkRefsHex(m_compiler->object(contract)).empty())
+			uncompiledContracts.insert(contract.substr(contract.find_last_of('/') + 1));
+
+	if (uncompiledContracts.size() == m_compiler->contractNames().size())
+		sout() << "Input did not contain any contracts to compile." << endl;
+	else
+		for (auto const& uncompiledContract: uncompiledContracts)
```

Comment:
I don't think we need the `else` here. Having interfaces and abstract contracts is **very** common. We don't want to always list them. We only want to prompt the user when all contracts in the project are like this and the `if` already covers that.

I'd also adjust the message to be clearer:
```suggestion
	if (uncompiledContracts.size() == m_compiler->contractNames().size())
	{
		sout() << "Input does not contain any deployable contracts. ";
		sout() << "Note that interface and abstract contract definitions do not produce bytecode." << endl;
	}
```

Alternatively, you could have an `if` that prints that there were no contracts if `m_compiler->contractNames().empty()` and that there were no deployable contracts otherwise (with the note about interfaces/abstract contracts).

You could also say how many non-deployable contracts were actually found but that's just extra.